pkg/util/log/log.go

```diff
@@ -19,13 +21,14 @@ var (
 	// Logger is a shared go-kit logger.
 	// TODO: Change all components to take a non-global logger via their constructors.
 	// Prefer accepting a non-global logger as an argument.
-	Logger = log.NewNopLogger()
+	Logger         = log.NewNopLogger()
+	bufferedLogger *LineBufferedLogger
 )
 
 // InitLogger initialises the global gokit logger (util_log.Logger) and overrides the
 // default logger for the server.
-func InitLogger(cfg *server.Config) {
-	l := newBasicLogger(cfg.LogFormat)
+func InitLogger(cfg *server.Config, buffered bool, sync bool) {
```

Comment:
Is there any point in the `sync` flag, as it and `buffered` should be mutually exclusive? I.e., if both are true (never happens in the code base), the sync writer will replace the buffered writer.

Reply:
I just kept it in place to allow the old synchronous functionality to be used if desired.

Reply:
@dannykopping I mean, can't you achieve the same by passing `false` for `buffered`? I don't understand the point in having two mutually exclusive booleans.

Reply:
Buffered and sync are not mutually exclusive, so I just kept both of them.
I put this together very quickly to help you folks with an incident.
I'll implement it however you want, but I wanted to keep the old pattern in case you needed to flip back.

Reply:
Ah, reading it again, I can see that if you pass `true` for both `buffered` and `sync`, the writer will be a sync writer wrapping `bufferedLogger`. Does that make sense though? Would we ever want to do that?

To flip back to the old pattern, couldn't that just be handled by passing `false` for `buffered` (provided that there's no separate `sync` argument, and `false` for `buffered` would mean sync)?

Reply:
> Does that make sense though? Would we ever want to do that?

I did this purely for backwards-compatibility. I don't know why a synchronous writer was used in the first place, so I supposed a buffered + sync writer might make sense to whomever configured it this way. I can't see a reason for holding a lock on every log write since it's a double penalty (if you include the cost of a syscall) on an action that should be very fast.

Reply:
Chatting with Danny, we're in agreement that support for creating a combined sync+buffered logger isn't necessary. It's no big deal though, and can be cleaned up later (just an internal API issue).